test/integration/api-create.test.js

```diff
@@ -549,13 +588,38 @@ test('fabric setup', function (tt) {
         }
     });
 
+    tt.test('prefer full id over name', function (t) {
+        // fNetwork3 is named with the full dockerId of fNetwork1.
+
+        h.createDockerContainer({
+            vmapiClient: VMAPI,
+            dockerClient: DOCKER_ALICE,
+            test: t,
+            extra: { 'HostConfig.NetworkMode': fNetwork3.name },
+            start: true
+        }, oncreate);
+
+        function oncreate(err, result) {
+            var nics = result.vm.nics;
+            t.equal(nics.length, 1, 'only one nic');
+            t.equal(nics[0].network_uuid, fNetwork1.uuid, 'correct network');
```

Comment:
This is the pathological case where a user has taken the `id` of `fNetwork1` and used it as the `name` of `fNetwork3`. So if the user specifies that same string, docker (and hence us) should pick the network whose `id` matches, not the name.